protocol/contracts/beanstalk/init/reseed/L2/ReseedSilo.sol

```diff
@@ -70,114 +63,82 @@ contract ReseedSilo {
-        reseedSiloDeposit(beanDeposits);
-
-        // initialize beanEthDeposits.
-        reseedSiloDeposit(beanEthDeposits);
-
-        // initialize beanWstEthDeposits.
-        reseedSiloDeposit(beanWstEthDeposits);
-
-        // initialize beanStableDeposits.
-        reseedSiloDeposit(bean3CrvDeposits);
-
-        // initialize urBeanDeposits.
-        reseedSiloDeposit(urBeanDeposits);
-
-        // initialize urBeanLpDeposits.
-        reseedSiloDeposit(urBeanLpDeposits);
+        // initialize deposits.
+        reseedSiloDeposit(deposits);
     }
 
     /**
-     * @notice reseed the silo deposit for a given token.
-     * @param siloDeposit The silo deposit data
+     * @notice reseed the silo deposits. 
+     * @param  deposits an array of account deposits for any token
+     * where each account's deposits can have many entries for the same token.
      * @dev all deposits and accounts are mown to the current season.
      */
-    function reseedSiloDeposit(SiloDeposits calldata siloDeposit) internal {
-        uint256 totalCalcDeposited;
-        uint256 totalCalcDepositedBdv;
-        uint256 stalkIssuedPerBdv = s.sys.silo.assetSettings[siloDeposit.token].stalkIssuedPerBdv;
-        for (uint256 i; i < siloDeposit.siloDepositsAccount.length; i++) {
-            AccountSiloDeposits memory deposits = siloDeposit.siloDepositsAccount[i];
-            uint128 totalBdvForAccount;
-            uint256 accountStalk;
-            for (uint256 j; j < deposits.dd.length; j++) {
+    function reseedSiloDeposit(AccountSiloDeposits[] calldata deposits) internal {
+        uint256 stalkIssuedPerBdv;
+        // for all deposits of a token and account.
+        for (uint256 i; i < deposits.length; i++) {
+            uint128 calculatedBdv;
+            uint256 calculatedStalk;
+            // different stalkIssuedPerBdv for each token in the AccountSiloDeposits struct.
+            stalkIssuedPerBdv = s.sys.silo.assetSettings[deposits[i].token].stalkIssuedPerBdv;
+            // for all of account's deposits.
+            for (uint256 j; j < deposits[i].dd.length; j++) {
+                // get stem from depositId.
+                (, int96 stem) = LibBytes.unpackAddressAndStem(deposits[i].dd[j].depositId);
                 // verify that depositId is valid.
-                int96 stem = deposits.dd[j].stem;
-                require(siloDeposit.stemTip >= stem, "ReseedSilo: INVALID_STEM");
-                uint256 depositId = LibBytes.packAddressAndStem(siloDeposit.token, stem);
-
-                // add deposit to account. Add to depositIdList.
-                s.accts[deposits.accounts].deposits[depositId].amount = deposits.dd[j].amount;
-                s.accts[deposits.accounts].deposits[depositId].bdv = deposits.dd[j].bdv;
-                s.accts[deposits.accounts].depositIdList[siloDeposit.token].depositIds.push(
-                    depositId
+                require(deposits[i].stemTip >= stem, "ReseedSilo: INVALID_STEM");
```

Comment:
think we can remove this, but defer to you on the utility (given that depositId and stemTip are inputs here

Reply:
Agreed. Since we re minimizing logic this isn't really necessary.  Fixed here: https://github.com/BeanstalkFarms/Beanstalk/pull/1002/commits/cd4ac34303bc1e6ba4ceca7b369ef79416ac597a